### PR TITLE
[WIP] Allow DDIMInverseScheduler to use same number of noising and denoising steps

### PR DESCRIPTION
**Pull request by kathath**
For the final DDIMScheduler step, we are already at t=0 and are predicting the sample at
t=0-num_train_steps/num_inference_steps.
The current DDIMInverseScheduler implemetation starts with the original image at t=0. But to match DDIMScheduler, one needs to start at t=0-num_train_steps/num_inference_steps in order to revert all denoising steps.
This can also be seen by the fact that there are num_inference_steps denoising steps but only num_inference_steps-1 noising steps in StableDiffusionPix2PixZeroPipeline and StableDiffusionDiffEditPipeline (those two pipelines are currently using DDIMInverseScheduler).

In contrast to the current DDIMInverseScheduler implementation, in [Null-text Inversion for Editing Real Images using Guided Diffusion Models](https://github.com/google/prompt-to-prompt/#null-text-inversion-for-editing-real-images) inversion starts at t=0-num_train_steps/num_inference_steps and reverses all denoising steps.
In the description of [DDIMInverseScheduler](https://huggingface.co/docs/diffusers/v0.13.0/en/api/schedulers/ddim_inverse), it is stated that "The implementation is mostly based on the DDIM inversion definition of [Null-text Inversion for Editing Real Images using Guided Diffusion Models](https://arxiv.org/pdf/2211.09794.pdf)" - which is currently not quite the case.

This PR allows to use DDIM inversion as implemented in  [Null-text Inversion for Editing Real Images using Guided Diffusion Models](https://github.com/google/prompt-to-prompt/#null-text-inversion-for-editing-real-images), when setting revert_all_steps=True.

This PR is strongly influenced by [https://github.com/google/prompt-to-prompt/#null-text-inversion-for-editing-real-images](https://github.com/google/prompt-to-prompt/#null-text-inversion-for-editing-real-images).

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hmm @clarencechen wdyt here? It essentially reverts the PR here: https://github.com/huggingface/diffusers/pull/2619

**Comment by clarencechen** [comment]
@patrickvonplaten I wouldn't say it reverts #2619, since the `DDIMInverseScheduler` implementation before then was also dropping the last (most noisy) timestep, but it does revert the semantic interpretation of the `timestamp` argument away from "timestamp corresponding to current noise level of input", which I personally strongly prefer. I think I would wait until a fine-tuned model addressing #3475 is out, since the current timestep I'm skipping corresponds to the final `steps_offset` training timesteps of the `DDIMScheduler` denoising process.

File: src/diffusers/schedulers/scheduling_ddim_inverse.py [hunk 1]
```diff
@@ -99,18 +99,21 @@ class DDIMInverseScheduler(SchedulerMixin, ConfigMixin):
             option to clip predicted sample for numerical stability.
         clip_sample_range (`float`, default `1.0`):
             the maximum magnitude for sample clipping. Valid only when `clip_sample=True`.
-        set_alpha_to_zero (`bool`, default `True`):
-            each diffusion step uses the value of alphas product at that step and at the previous one. For the final
-            step there is no previous alpha. When this option is `True` the previous alpha product is fixed to `0`,
-            otherwise it uses the value of alpha at step `num_train_timesteps - 1`.
+        set_alpha_to_one (`bool`, default `True`):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddim_inverse.py:102]: Why do we have to do the renaming here? This is backwards breaking IMO

File: src/diffusers/schedulers/scheduling_ddim_inverse.py [hunk 2]
```diff
@@ -124,18 +127,13 @@ def __init__(
         beta_schedule: str = "linear",
         trained_betas: Optional[Union[np.ndarray, List[float]]] = None,
         clip_sample: bool = True,
-        set_alpha_to_zero: bool = True,
+        set_alpha_to_one: bool = True,
         steps_offset: int = 0,
         prediction_type: str = "epsilon",
         clip_sample_range: float = 1.0,
+        revert_all_steps: bool = False,
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddim_inverse.py:134]: is this ever used?

**Review by patrickvonplaten — COMMENTED** [review]
Left some comments inline

**Pull request closed without merging**